dts/arm/nordic/nrf5340_cpuappns.dtsi

```diff
@@ -48,6 +48,24 @@
 			 */
 			#include "nrf5340_cpuapp_peripherals.dtsi"
 		};
+
+		/* Cryptocell is a secure-only peripheral, but the availability
+		 * of this peripheral and the cryptocell variant is still useful
+		 * information for the non-secure application as it can access
+		 * crypto features through secure-services.
+		 */
+		cryptocell: crypto@50844000 {
+			compatible = "nordic,nrf-cc312";
+			reg = <0x50844000 0x1000>;
+			status = "okay";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			cryptocell312: crypto@50845000 {
+				compatible = "arm,cryptocell-312";
+				reg = <0x50845000 0x1000>;
+				interrupts = <68 NRF_DEFAULT_IRQ_PRIORITY>;
+			};
```

Comment:
I think this needs a better explanation. Any links to the reference manual?

Reply:
Which reference manual are you looking for?
I am using the same node entry as exists before: https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/nordic/nrf91.dtsi#L49

If that is wrong then it can be changed, but I don't think that should be part of this review.
I'm only making sure that the node is available for both boards so that we don't have to have workarounds for NCS for this case.

There is currently a noup patch that I'm trying to improve: https://github.com/nrfconnect/sdk-zephyr/pull/1253